march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -249,18 +253,37 @@ void MarchHardwareInterface::write(ros::Duration elapsed_time)
 {
   positionJointSoftLimitsInterface.enforceLimits(elapsed_time);
 
+  ROS_DEBUG("Before limits: Trying to actuate joint %s, to %lf "
```

Comment:
don't you want this not loop through all the joints like the after limits ROS_debug message?